src/transport/xqc_cid.c

```diff
@@ -250,7 +250,7 @@ xqc_int_t
 xqc_cid_switch_to_next_state(xqc_cid_set_t *cid_set, xqc_cid_inner_t *cid, xqc_cid_state_t next_state)
 {
     if (xqc_cid_in_cid_set(cid_set, &cid->cid) == NULL) {
-        return -XQC_ECONN_CID_NOT_FOUND;
+        return XQC_OK;
```

Comment:
Should not hide the error inside the function

Reply:
About solve the issue with CID retirement retransmission, we need to check the state before switch the CID state, instead of letting the wrong state pass.

Reply:
fixed by validating CID after receiving a RC frame